website/docs/guides/container_service_extension.html.markdown

```diff
@@ -0,0 +1,438 @@
+  network_pool_name = "NSX-T Overlay"
+  provider_vdc_name = "nsxTPvdc1"
+
+  compute_capacity {
+    cpu {
+      limit = 0
+    }
+
+    memory {
+      limit = 0
+    }
+  }
+
+  storage_profile {
+    name    = "*"
+    enabled = true
+    limit   = 0
+    default = true
+  }
+
+  enabled                  = true
+  enable_thin_provisioning = true
+  enable_fast_provisioning = true
+  delete_force             = true
+  delete_recursive         = true
+}
+```
+
+### Step 2: Configure networking
+
+For the Kubernetes clusters to be functional, you need to provide some networking resources to the target VDC:
+
+* [Tier-0 Gateway](/providers/vmware/vcd/latest/docs/resources/external_network_v2)
+* [Edge Gateway](/providers/vmware/vcd/latest/docs/resources/nsxt_edgegateway)
+* [Routed Network](/providers/vmware/vcd/latest/docs/resources/network_routed_v2)
+* [SNAT rule](/providers/vmware/vcd/latest/docs/resources/nsxt_nat_rule)
+
+The [Tier-0 Gateway](/providers/vmware/vcd/latest/docs/resources/external_network_v2) will provide access to the
+outside world. For example, this will allow cluster users to communicate with Kubernetes API server through `kubectl` and
+download required dependencies for the cluster to be functional.
+
+Here is an example on how to configure this resource:
+
+```hcl
+data "vcd_nsxt_manager" "main" {
+  name = "my-nsxt-manager"
+}
+
+data "vcd_nsxt_tier0_router" "router" {
+  name            = "VCD T0 edgeCluster"
+  nsxt_manager_id = data.vcd_nsxt_manager.main.id
+}
+
+resource "vcd_external_network_v2" "cse_external_network_nsxt" {
+  name        = "extnet-cse"
+  description = "NSX-T backed network for k8s clusters"
+
+  nsxt_network {
+    nsxt_manager_id      = data.vcd_nsxt_manager.main.id
+    nsxt_tier0_router_id = data.vcd_nsxt_tier0_router.router.id
+  }
+
+  ip_scope {
+    gateway       = "88.88.88.1"
+    prefix_length = "24"
+
+    static_ip_pool {
+      start_address = "88.88.88.88"
+      end_address   = "88.88.88.100"
+    }
+  }
+}
+```
+
+Create also an [Edge Gateway](/providers/vmware/vcd/latest/docs/resources/nsxt_edgegateway) that will use the recently created
+external network. This will act as the main router connecting our nodes in the internal network to the external (Tier 0 Gateway) network:
+
+```hcl
+resource "vcd_nsxt_edgegateway" "cse_egw" {
+  org      = vcd_org.cse_org.name
+  owner_id = vcd_org_vdc.cse_vdc.id
+
+  name                = "cse-egw"
+  description         = "CSE edge gateway"
+  external_network_id = vcd_external_network_v2.cse_external_network_nsxt.id
+
+  subnet {
+    gateway       = "88.88.88.1"
+    prefix_length = "24"
+    primary_ip    = "88.88.88.88"
+    allocated_ips {
+      start_address = "88.88.88.88"
+      end_address   = "88.88.88.100"
+    }
+  }
+  depends_on = [vcd_org_vdc.cse_vdc]
+}
+```
+
+This will create a basic Edge Gateway, you can of course add more complex [firewall rules](/providers/vmware/vcd/latest/docs/resources/nsxt_firewall)
```

Comment:
If you think is appropriate I can include an example of firewall. I still don't have clear if we need also to include the new `vcd_nsxt_route_advertisement` resource here.

Reply:
CC: @Didainius @vbauzysvmware for input